### ISSUE TITLE
Is a new release happening soon?

### ISSUE THREAD

**Issue by SuperSandro2000** [Exiv2/exiv2#2406]
exiv2 has currently some medium to high opne CVEs in the latest release (like this 9.8 https://nvd.nist.gov/vuln/detail/CVE-2022-3717 ) which should be patched rather quickly in distros and software distributions. Normally applying a patch for the fixes is rather easy but because of the big formatting patches (see https://github.com/Exiv2/exiv2/blob/main/.git-blame-ignore-revs) the patches/commits no longer apply on the latest release.

Are there any plans for a new release in the new future? If a release is far in the future maybe the patches could be rebased on the latest release and a new security fixes only release could be done?

**Comment by risicle** [comment]
I'd like to add to this a bit of feedback on having *actionable* vulnerability reports. Regarding the 6 recently released CVEs:

https://nvd.nist.gov/vuln/detail/CVE-2022-3717
https://nvd.nist.gov/vuln/detail/CVE-2022-3718
https://nvd.nist.gov/vuln/detail/CVE-2022-3719
https://nvd.nist.gov/vuln/detail/CVE-2022-3755
https://nvd.nist.gov/vuln/detail/CVE-2022-3756
https://nvd.nist.gov/vuln/detail/CVE-2022-3757

For each of these the suggested course of action is patching, but not a single one of the indicated patches apply to the most recent stable release without significant alteration. Any such alteration would also be a rather blind process because the proof of concepts I could find for these issues don't seem to trigger v0.27.5, so it would be hard to verify that it's been fixed.

It's possible these aren't triggering v0.27.5 because of a recent slightly mysterious refactor of the BMFF support that was done in August: https://github.com/Exiv2/exiv2/commit/3456f3098897330ea3a3efff9b19f624e407c22e which wasn't accompanied by any announcement.

CVE-2022-3717 is of special note because I don't think it affected any versions outside _that PR_, so I'm not quite sure what the purpose of that was.

So it's difficult for a distributor/user to know what to do now. If the answer is "run an unstable version" (which right now looks like it might be the only sensible option for shipping a safe exiv2), then you may as well make an immediate release because I guess at this point any QA criteria go out the window.

**Comment by postscript-dev** [comment]
> Are there any plans for a new release in the new future?

After Robin retired from Exiv2, @nehaljwani kindly offered to take over release engineering. The transition is still relatively new and will take time to work smoothly.

As far as I know, @nehaljwani doesn't follow the day-to-day posts but can be contacted by mentioning him.

**Comment by clanmills** [comment]
I will help @nehaljwani and anybody to make a release.  I'm pleased to be retired and will not return to the project.  I am happy to see Exiv2 move forward and will help when asked.

**Comment by clanmills** [comment]
@nehaljwani and I are meeting today (2022-11-12) on Google Meet to discuss this.  All welcome.

Robin/Nehal
Saturday, 12 November · 14:00 – 15:00 UTC
Google Meet joining info
Video call link: https://meet.google.com/mfk-zpii-gic

If you want an invitation, email me: at robinwmills@gmail.com

**Comment by clanmills** [comment]
The most important matter in this release is the scope and version.  Choices:

1.  Release 'main' as version 0.28.0 (or possibly 0.90.0 to make it very clear that this is not a simple successor to 0.27)
2. Release 'main' as version 1.00.0 (I don't think 'main' is v1.00.0)
3. Patch v0.27.5 with the security fixes and release v0.27.6

I don't know the status of the 0.27-maintenance branch.  If security fixes have to be back-ported to branch 0.27-maintenace, the work involved could be considerable.

**Comment by nehaljwani** [comment]
Hello everyone! Robin has shared the steps involved in cutting a release.

IIUC, the `main` branch diverged from `0.27-maintenance` right after 0.27.4 with [this](https://github.com/Exiv2/exiv2/commit/350b7b0808673bff28de6c6e89263c7a782b633d) commit.

If @kevinbackhouse can confirm that the required security fixes are present in the 0.27-maintainance branch (or not required because of irrelevance), we can proceed with a v0.27.6. As for a release from the main branch, I vote for v0.75.0.  

I invite @postscript-dev, @kmilos, @piponazo, @neheb to voice their opinion.

I'm hoping to cut the releases by the end of this month with the versions listed above, if no concerns are raised by then.

**Comment by postscript-dev** [comment]
@nehaljwani 
Thanks for taking the time to work on this. As I build from source, the new release doesn't affect me much.

I had a couple of thoughts on the release process.

I recently spotted that files used when building the website are duplicated (e.g., https://github.com/Exiv2/exiv2/blob/main/doc/templates/Makefile and https://github.com/Exiv2/team/blob/main/website/Makefile). As far as I know, the Exiv2/team ones are not being kept up to date. There are some differences in Exiv2/Exiv2/docs/templates - particularly on the `main` branch.

Also on the `main` branch, the manpage has changed to markdown format (i.e.[Exiv2/Exiv2/man/man1/exiv2.1](https://github.com/Exiv2/exiv2/blob/main/man/man1/exiv2.1) to  [Exiv2/Exiv2/exiv2.md](https://github.com/Exiv2/exiv2/blob/main/exiv2.md)). As a result, it wont be automatically added to https://exiv2.org/manpage.html. The manpage text may also need minor reformatting to fit any line limits used by the postscript file. If you need help with the manpage text, then let me know.

It is worth noting that the `main` branch has diverged a *LOT* from the commit that you named. For 0.75.0 , apart from the size of the release notes, there could be other minor changes needed to the release process. As a release is needed, perhaps it would be better to complete 0.27.6 first and then work on 0.75.0 after.

**Comment by piponazo** [comment]
Hi everybody!
I have not been able to contribute much to the project lately, but I still keep to keep an eye on what's going on and deal with the CI problems when it is needed. Since winter is coming and is full of terrors, I will probably show up more here 😄 .

For me it also sounds like a good plan to make 2 releases:
* 0.27.6 
* 0.28 or 0.75 (I do not have a strong preference for or or the other)

Whenever we think that `main` is in a state mature enough we can think about 1.0.

**Comment by kevinbackhouse** [comment]
Sorry, I haven't been paying to this issue. I'm quite angry to discover that a bunch of CVEs have been filed without consulting any of us. (I very much doubt it was done by anybody on the Exiv2 team, because we would have used GitHub Security advisories instead.) I haven't checked them all yet, but I'm pretty confident that all of those CVEs are bogus because they were introduced on the development branch (`main`) and fixed within a few weeks. As an example, CVE-2022-3717, which is cited as the main example in this issue, was fixed as the [8th commit](https://github.com/Exiv2/exiv2/pull/2381/commits) in #2381, so it never even got merged into the main branch.

We have a [security policy](https://github.com/Exiv2/exiv2/security/policy) which spells our very clearly that bugs on the main branch are not security bugs. Only bugs in official releases, such as [v0.27.5](https://github.com/Exiv2/exiv2/releases/tag/v0.27.5) are potential security vulnerabilities. I have been paying attention to all the potential security issues and I'm pretty confident that none of the bugs found recently have been reproducible on the [0.27-maintenance](https://github.com/Exiv2/exiv2/tree/0.27-maintenance) branch.

I will contact the CNA to dispute those CVEs and get them removed.

**Comment by kevinbackhouse** [comment]
Regarding a new release, I am in favor of doing it soon. I think we should do a v0.27.6 and I think it's time to do a v1.0.0 too. My own main goal for 1.0.0 was to replace all the uses of the `long` type with something more appropriate, such as `size_t`. I think that's mostly done - I think there are still some uses in some of the `.cpp` files, but the header files are mostly clean now so the remaining problems can be fixed without affecting the public API.

**Comment by 1div0** [comment]
WRT v1.0.0 I do hope Big Blue Red Hat lawyers will at least state in https://bugzilla.redhat.com/show_bug.cgi?id=1979565 that your camera means your data and no fooced shitware patents apply.

**Comment by kmilos** [comment]
Thanks for stepping up @nehaljwani 👍 

If it plugs all the known security holes, I'm in favour of 0.27.6 asap, as it has some nice added functionality and bug fixes backported.

Re the next version, I don't really care what it's called, as long it comes soonish as well (there are projects depending on exiv2 that are slowly starting to require at least C++17). [One thing that needs to be sorted out though is the SO version mess](https://github.com/Exiv2/exiv2/issues/2020#issuecomment-990851749) (from that vantage point, I'm even ok w/ 0.28.0...)

**Comment by kevinbackhouse** [comment]
Those bogus CVEs have been rejected now. For example: https://nvd.nist.gov/vuln/detail/CVE-2022-3717

**Comment by postscript-dev** [comment]
@nehaljwani 
Support for video was removed from the `main` branch however, some users requested this to be restored. @mohamedchebbii has worked hard to recover and modernise the code to be compatible with C++17. It would be good to include this in the new `main` release, if possible.  @mohamedchebbii, how close is your work to being merged?

**Comment by mohamedchebbii** [comment]
Hello @postscript-dev ,
I pushed my three pull requests and they seems good for me:
https://github.com/Exiv2/exiv2/pull/2413
https://github.com/Exiv2/exiv2/pull/2415
https://github.com/Exiv2/exiv2/pull/2416

Please let me know, If I need to fix or add somthing else ?
Regards,

**Comment by postscript-dev** [comment]
@mohamedchebbii 
Thanks for the update. As I don't have much free time, I was planning to leave your review to someone else.

**Comment by benmccann** [comment]
The video support has been merged. Since it's a major feature, maybe the next release should be 0.28.0? It would be great to be able to get it out there. We're stuck on an outdated version of exiv2 (the last one with video support before it was removed), and are really looking forward to upgrading soon :smile: Thanks to @mohamedchebbii for all the great work on that feature and thanks to @piponazo for the reviews!

**Comment by kevinbackhouse** [comment]
As I just commented in #2450, I would like to roll back the recently merged video support until it has been better tested.

I think Exiv2 was in very good shape before those PRs were merged, so my suggestion is that we start preparing 0.27.6 and 1.0.0 releases based on the state of the code as it was on [2022-12-30](https://github.com/Exiv2/exiv2/commit/9ca161d1e5696623a2a3b97860458f90ed01c511).

**Comment by postscript-dev** [comment]
@nehaljwani 
I was reading some closed issues and was reminded of something. In regards to a new `main` release, we now require a minimum of Windows 10 for the command line programs (#2090). This applies to the exiv2 app and sample programs that we include in the download. This caveat needs to be added to the Exiv2 website as part of the release.

**Comment by neheb** [comment]
According to https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/setlocale-wsetlocale?view=msvc-170 , prior Windows versions can be supported if libc is statically linked.

**Comment by nehaljwani** [comment]
**Exiv2 v0.27.6 2023-01-TBD**

| Topic          | More Info |
|:--             |:--        | 
| Remaining ( 0) | https://github.com/Exiv2/exiv2/milestone/10 |
| Completed (41) | https://github.com/Exiv2/exiv2/milestone/10?closed=1 |

**Exiv2 v0.27.6 Acknowledgements**
```
~/r/exiv2 ❯❯❯ git shortlog -s -n v0.27.5..HEAD
    30  Miloš Komarčević
    20  Robin Mills
    13  Luis Díaz Más
    11  Kevin Backhouse
     4  Luis Diaz
     3  Peter
     1  Christoph Hasse
```

**Exiv2 v0.27.6 Release Notes (updated 2023-01-16)**

|  Group        | PR     |  Topic                                        | Issue  |
|:---           |:--     |:--                                            |:--     |
| **Documentation** | #2404 | Update docs concerning BMFF-based files | #2398
|               | #2328 | Fix rename text in manpage | #2290
| **Lens**          | #2315 | Add Nikon3.WhiteBalanceBias2 |
|               | #2291 | Add Nikon LensData v0802 |
|               | #2243 | Add some F mount lenses | #2193, #2196, #2214
|               | #2167 | Initial support for OM System MakerNote | #2126
|               | #2046 | Add Sony ARW compression to dict |
| **Build**         | #2315 | Upgrade conan to version 1.51.0
|               | #2241 | Fix CI jobs - Update conan packages
|               | #2142 | Update CI actions
|               | #2140 | Update Windows CI workflows
|               | #2033 | Updating documentation to respect ctest
|               | #2032 | Switch Cygwin CI to GHA
|               | #2025 | Nightly pre-builds for 0.27 | #1984
|               | #2024 | Running tests with ctest | #2022
|               | #2019 | Switch MinGW CI to GHA
|               | #2018 | Fix conan ci
|               | #2004 | Streamline MinGW package installation for CI
|               | #2003 | Fix broken Cygwin/CI
|               | #2002 | Fix CMake build type | #1558
| **Bugs & Fixes**  | #2382 | Exif start can be at any byte in payload, not word aligned | #2381
|               | #2375 | Fix exception type when writing BMFF files | #2350
|               | #2333 | Add more MIME type mappings for TIFF-based raws | #2260
|               | #2261 | Fix naming of canon EF 35-80mm | #2247
|               | #2269 | Replace assert with enforce | #2268
|               | #2256 | PNG: always strip the existing iCCP chunk |
|               | #2239 | Account for header bytes for Exif and XMP boxes | #2233
|               | #2194 | Fix Integer overflow in Photoshop::setIptcIrb | #2179
|               | #2192 | Fix Integer-overflow in sumToLong | #2190
|               | #2186 | Fix out of bounds read in isValidBoxFileType() | #2178
|               | #2153 | Fix in Jp2 metadata writing & improvements in reading | #2147
|               | #2139 | Strip XMP raw packet before decoding | #2126
|               | #2045 | Add tiff tags | #2044
|               | #2026 | Add more DNG 1.6 tags
|               | #2037 | Fix bug in iterating over the elements of dateStrings | #2036
|               | #2030 | Use memmove in TiffEncoder::updateDirEntry | #2027
|               | #2016 | Treat Exif.Sony1.PreviewImage as undefined tag | #2001
| **Tests**         | #2269 | Regression test for Exiv2#2268 | #2268

If I've failed to acknowledge anyone's contribution, I apologize. Please let me know and I'll update this comment.

**Comment by kmilos** [comment]
Thanks! @nehaljwani 

Since this is also driven by security needs, maybe it's worth listing what CVEs have been plugged in the release notes as well? @kevinbackhouse

**Comment by kevinbackhouse** [comment]
> Thanks! @nehaljwani
> 
> Since this is also driven by security needs, maybe it's worth listing what CVEs have been plugged in the release notes as well? @kevinbackhouse

No new CVEs in this release. All known security bugs were caught during development and never made it into an official release.

**Comment by kmilos** [comment]
> All known security bugs were caught during development and never made it into an official release.

I just re-read your response a while back - so no valid CVEs against 0.27.5, and the release notes look good then. 👍

**Comment by nehaljwani** [comment]
Dear folks, Exiv2 v0.27.6 has been released!

>... 1.0.0 releases based on the state of the code as it was on [2022-12-30](https://github.com/Exiv2/exiv2/commit/9ca161d1e5696623a2a3b97860458f90ed01c511).

I'll start working on the `v1.0.0` major release based on https://github.com/Exiv2/exiv2/commit/9ca161d1e5696623a2a3b97860458f90ed01c511

Please raise concerns, if any.

**Comment by kmilos** [comment]
> I'll start working on the `v1.0.0` major release based on [9ca161d](https://github.com/Exiv2/exiv2/commit/9ca161d1e5696623a2a3b97860458f90ed01c511)
> 
> Please raise concerns, if any.

See again https://github.com/Exiv2/exiv2/issues/2406#issuecomment-1322291596

**Comment by jim-easterbrook** [comment]
Do you plan to add a `Source` download to the `Linux`, `Darwin` and `msvc`? The `Download Source` button on the exiv2.org download page is currently broken.

**Comment by postscript-dev** [comment]
@nehaljwani 
I noticed that the release notes for `0.27.6` are missing on https://exiv2.org/whatsnew.html . Hopefully a simple fix.

**Comment by 1div0** [comment]
@nehaljwani Thank you so much.

So far so good.

While experimenting with the various build configuration by enebling NLS support, I have noticed the following:

`
 CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
   Failed to find Gettext libintl (missing: Intl_LIBRARY)
 Call Stack (most recent call first):
   /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
   /usr/share/cmake/Modules/FindIntl.cmake:161 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
   cmake/findDependencies.cmake:70 (find_package)
   CMakeLists.txt:77 (include)

 Configuring incomplete, errors occurred!
 See also "/2TB/usr/src/github.com/1div0/exiv2/Linux/x86-64/CMakeFiles/CMakeOutput.log".
 See also "/2TB/usr/src/github.com/1div0/exiv2/Linux/x86-64/CMakeFiles/CMakeError.log".
`
This is on F37, everything up to date. Is it already known issue?

My free(time_t) is very limited, will get back to topic by the WE.

**Comment by jim-easterbrook** [comment]
> The `Download Source` button on the exiv2.org download page is currently broken.

It's OK now! Thanks for the quick fix.

**Comment by benmccann** [comment]
Will 1.0 and 0.27.6 be essentially the same release? If so, what's the point of having the same release with two different version numbers? Maybe 1.0 should include some big exciting new features like video support?

**Comment by SuperSandro2000** [comment]
> Failed to find Gettext libintl (missing: Intl_LIBRARY)

Missing gettext I would guess.



> If so, what's the point of having the same release with two different version numbers?

To promote a stable release and know it is stable. An alternative would be to do a beta or rc candidate first.

**Comment by benmccann** [comment]
If we want to treat 0.27.6 as a beta or RC, maybe we'd want to let more time pass so that we're able to collect some bug reports?

**Comment by kmilos** [comment]
> If we want to treat 0.27.6 as a beta or RC, maybe we'd want to let more time pass so that we're able to collect some bug reports?

We don't want to do that. The main branch (from which "1.0.0" or whatever-the-next-version-number will be derived), has significant changes to the codebase and is to be considered very different from the 0.27.x series (although feature-wise, indeed similar at this point in time).

**Comment by 1div0** [comment]
@kmilos Can't agree more. For one divided by zero release there needs to be clear way forward regarding the BMFF. For all fancy new formats as well as renewed video support.

**Comment by clanmills** [comment]
Well done, @nehaljwani, for the 0.27.6 release on 2023-01-18, which was my 72nd birthday.  Thank You, Nehal and everybody, for getting this done.

**Comment by 1div0** [comment]
> > Failed to find Gettext libintl (missing: Intl_LIBRARY)
> 
> Missing gettext I would guess.
> 

/usr/share/cmake/Modules/FindIntl.cmake

.. note::
  On some platforms, such as Linux with GNU libc, the gettext
  functions are present in the C standard library and libintl
  is not required.  ``Intl_LIBRARIES`` will be empty in this
  case.

This seems to be the case.

F37 now, I will check RHEL and Rocky ASAP.

**Comment by 1div0** [comment]
### Rocky Linux 8.7 (Green Obsidian)

[peter.kovar@xn--dsseldorf-q9a Test]$ LANG=sk_SK.UTF-8 exiv2 *
IMGP0191.jxl         Názov súboru    : IMGP0191.jxl
IMGP0191.jxl         Veľkosť súboru  : 684603 bajtov
IMGP0191.jxl         MIME typ        : image/generic
IMGP0191.jxl         Veľkosť obrázka : 3040 x 2024
IMGP0191.jxl         Náhľad          : Nič
IMGP0191.jxl         Zmačka fotoaparátu: PENTAX Corporation 
IMGP0191.jxl         Model fotoaparátu: PENTAX K100D       
IMGP0191.jxl         Časová známka obrázka: 2019:09:21 21:41:07
IMGP0191.jxl         File number     : 
IMGP0191.jxl         Čas expozície   : 0.4 s
IMGP0191.jxl         Clona           : F3.5
IMGP0191.jxl         Skreslenie expozície: 0 EV
IMGP0191.jxl         Blesk           : No, compulsory
IMGP0191.jxl         Skreslenie blesku: 
IMGP0191.jxl         Ohnisková vzdialenosť: 18.0 mm
IMGP0191.jxl         vzdialenosť subjektu: 
IMGP0191.jxl         ISO rýchlosť    : 800
IMGP0191.jxl         Režim expozície : nedefinované
IMGP0191.jxl         merací režim    : Multi-segment
IMGP0191.jxl         režim makro     : 
IMGP0191.jxl         Kvalita obrázka : 
IMGP0191.jxl         Vyváženie bielej: Auto
IMGP0191.jxl         Copyright       : 
IMGP0191.jxl         Komentár EXIF   : 

Almost perfect, with the one small exception: MIME type is reported as **image/generic** instead of the **image/jxl**.

@kmilos ?

**Comment by 1div0** [comment]
@kmilos it is correct in the **main**, so let's focus on 1.0 release.

**Comment by postscript-dev** [comment]
> 1div0 wrote: For one divided by zero release there needs to be clear way forward regarding the BMFF. For all fancy new formats as well as renewed video support.

I don't use BMFF files and haven't studied the legal arguments in detail.

However, I agree that resolving the BMFF issue would help Exiv2 move forward. One way to achieve this would be to join the Open Invention Network (OIN) - as discussed in #1447. In the first post, there is a list of benefits to joining but I think the following reasons also apply.
1. We can bring an end to the BMFF issue and provide a common way forward.
2. Once the BMFF issue is resolved, we are free to concentrate on other issues.
3. After joining, we can fully enable BMFF helping many  distributions, projects and users.

If there is agreement amongst the Exiv2 team, then I think joining only takes one of the maintainers a few minutes to sign.

After joining, the Exiv2 code could then be simplified on the `main` branch to enable BMFF for everyone. If joining is agreed upon quickly, we might even be able to include this in the next release.

It would be helpful to know what everone else thinks about this. Further discussions on joining the OIN can take place in #1447.

**Comment by kmilos** [comment]
> It would be helpful to know what everone else thinks about this.

I'd just simply drop this - Exiv2 already had BMFF parsing code in `jp2image` **_and_** `quicktimevideo` for ages.

**Comment by kmilos** [comment]
@1div0 It was actually [your patch](https://github.com/Exiv2/exiv2/pull/1824) that didn't get back-ported 😞

**Comment by 1div0** [comment]
I am so sorry about that. Really, multitasking is affecting me badly.

**Comment by kevinbackhouse** [comment]
> > I'll start working on the `v1.0.0` major release based on [9ca161d](https://github.com/Exiv2/exiv2/commit/9ca161d1e5696623a2a3b97860458f90ed01c511)
> > Please raise concerns, if any.
> 
> See again [#2406 (comment)](https://github.com/Exiv2/exiv2/issues/2406#issuecomment-1322291596)

This thread is quite long, so just chiming in to say that I agree with @kmilos that [9ca161d](https://github.com/Exiv2/exiv2/commit/9ca161d1e5696623a2a3b97860458f90ed01c511) would be a great candidate to create the `v1.0.0` release from.

**Comment by benmccann** [comment]
Video support has been merged. Would it be possible to get a new release which includes it?

**Comment by 1div0** [comment]
Thank you so much.

Everything is possible.

We need to make 120% sure there would not introduce any vulnerability by extensive testing on various large datasets.

Definitely doable.

**Comment by benmccann** [comment]
Yes, I definitely agree we should make sure video support and the library in general are well tested.

I do want to highlight that there's also risk in holding back a release for an extended period of testing. A lot of users are stuck on the last version of exiv2 with video support and can't upgrade until we do a release. E.g. all Synology NAS devices come with exiv2 0.27.3. Synology Photos uses it for photos and videos. There's a huge number of people stuck on older versions with vulnerabilities that have already been fixed. Until we release a new version they're not going to be able to upgrade to get any of the fixes. Even if we were to release with some vulnerabilities it would still be a huge improvement for a vast set of users and would greatly cut down on the security risks they're being exposed to. Hopefully we can make the testing period quick and release a new version without too much delay.

**Comment by 1div0** [comment]
Agree, but TANSTAAFL.

**Comment by kevinbackhouse** [comment]
Video is enabled by default and we have zero open bugs on OSS-Fuzz. 🥳
Would now be a good time to do a release?

**Comment by benmccann** [comment]
+1 for a new release! Thank you to both @mohamedchebbii and @kevinbackhouse for fixing the fuzzing issues!

**Comment by kmilos** [comment]
I still don't see a policy being decided on for the versioning.

**Comment by kevinbackhouse** [comment]
> I still don't see a policy being decided on for the versioning.

Is this the question of whether it should be 1.0.0 or 0.28.0? I have no opinion on that and don't mind either way.

**Comment by kmilos** [comment]
> Is this the question of whether it should be 1.0.0 or 0.28.0? I have no opinion on that and don't mind either way.

Not so much the number (I don't care much either), but the fact that the library SOVERSION is broken on main if we go for 1.0.0.

**Comment by 1div0** [comment]
I have found few corner cases with exotic video formats, but still had no time_t to fix these.

**Comment by nehaljwani** [comment]
I agree that it is high time to cut a release. I'll be able to put in the time this week and work on it.

>Video is enabled by default and we have zero open bugs on OSS-Fuzz. 🥳

As for the 1.0.0 vs 0.28.0 debate, my vote still goes for 1.0.0.

**Comment by neheb** [comment]
There’s https://github.com/Exiv2/exiv2/pull/2581 which is either a problem or just cosmetic.

**Comment by benmccann** [comment]
It seems cosmetic to me. I'm not sure we should block a release because there's a PR where the tests are failing. There's no indication there's any problem in `main`.

**Comment by benmccann** [comment]
> There’s https://github.com/Exiv2/exiv2/pull/2581 which is either a problem or just cosmetic.

https://github.com/Exiv2/exiv2/pull/2581 has been merged. Time for the release?

**Comment by kmilos** [comment]
> Time for the release?

The versioning has not be settled yet.

**Comment by benmccann** [comment]
@kevinbackhouse voted don't care: https://github.com/Exiv2/exiv2/issues/2406#issuecomment-1484948079
I will also throw in a vote for don't care

@nehaljwani voted for 1.0: https://github.com/Exiv2/exiv2/issues/2406#issuecomment-1485738464

I think unless anyone expresses a different opinion, we should go for 1.0 then as it's currently leading

How about voting with emojis? You can put one of the following reactions on this post:
- a :tada: if you want 0.28.0
- a :rocket: if you want 1.0
- a :heart: if you don't care

**Comment by kmilos** [comment]
Not talking about the number, but the mechanism. It's not fixed on main yet.

**Comment by benmccann** [comment]
Are we settled on the number at least? The mechanism only needs to be fixed if we go with 1.0, so we need to decide that before we know whether to tweak the mechanism

**Comment by benmccann** [comment]
Sure, but if we're going with 0.28.0, we don't need to block cutting a release on fixing `main`. Not that we shouldn't still fix `main`, but it's doesn't need to be a blocker

**Comment by kmilos** [comment]
Merged now 😉

**Comment by benmccann** [comment]
It looks like [the votes](https://github.com/Exiv2/exiv2/issues/2406#issuecomment-1511936787) are evenly split between 0.28.0 and 1.0 currently. Perhaps others would like to [weigh in](https://github.com/Exiv2/exiv2/issues/2406#issuecomment-1511936787) to break the tie?

**Comment by neheb** [comment]
https://0ver.org

**Comment by lectrode** [comment]
Random bystander here. As such I won't encroach on the actual version voting, but I would just like to say:

* The version scheme should support the desired frequency of releases (small or large) and should make that process quick and simple. For this project, it seems like it could benefit from smaller, more frequent releases, as waiting for a single large release may threaten to stall development (the longer the time between releases, the more difficult it is to vet all of the code changes between them, and the "scarier" the process of pushing a new version becomes)
* A new major version is often used to denote either compatibility breaking changes, or a new resolve / new philosophy governing a project. Per your discretion, this release could denote the latter (for example, a new resolution to release smaller, more frequent updates, or the transition from a sole primary developer to broader community development for long-term transformation and growth)
* While not strictly necessary, the bump from version 0.x.x to 1.x.x usually denotes that a project is no longer in beta/alpha state and can be regarded as stable enough for other projects to depend on. `exiv2` has met that criteria for quite some time now, so specifically denoting that wouldn't necessarily be required, but might be useful particularly for developers new to the open source ecosystem
* versioning schemes like "0ver" should not be adopted solely for the novelty - it should serve some function (i.e. ease of release, compatibility, etc), especially since with that scheme in particular you are restricting the use of the first digit and essentially making it meaningless

Good luck! Looking forward to the next release!

**Comment by 1div0** [comment]
1 dot zero/eu

**Comment by perdrix52** [comment]
PLEASE release a new release with the auto_ptr to unique_ptr changes ASAP.  My project is now converted to C++20 and I cannot use 27.6 .   My only other option would be to DL nightly build and I am very wary of doing that.

Thanks
David

**Comment by 1div0** [comment]
Better to avoid .0 release.

1.1.0 w/ BMFF & video.

**Comment by benmccann** [comment]
Perhaps the release process could be documented, or better yet automated, so that we don't place the burden on a single person? E.g. many of the JS projects I've worked on use [changesets](https://github.com/changesets/changesets) to help manage a changelog and then create a new release with a single button click on GitHub.

**Comment by benmccann** [comment]
It looks to me like the only thing we really need to do is to update the version number in `CMakeLists.txt`: https://github.com/Exiv2/exiv2/blob/5708bee1e68c7668ee685e41eb05b80dcc2414fb/CMakeLists.txt#L5

Is there anything else I'm missing? It's a bit weird that it's currently on 1.00.0.9 while we want to cut 0.28.0. Is that going to cause any issues?

It looks like the one other thing that's currently done when creating a new release is manually updating https://exiv2.org/whatsnew.html. Perhaps we should remove that page in favor of simply linking to https://github.com/Exiv2/exiv2/releases since it's just a manually created copy of it? That would simplify things and reduce the risk of them getting out-of-sync. It'd also reduce the risk of Google penalizing exiv2.org for duplicating content from another domain

**Comment by perdrix52** [comment]
Don't you mean the only thing to be done after fixing the bugs I reported you mean ... 

David

**Comment by krupkat** [comment]
> Don't you mean the only thing to be done after fixing the bugs I reported ...

Please be more polite to the developers.

**Comment by perdrix52** [comment]
I don't see that what I said was impolite - maybe it came over like that?  If so I apologise.

However at least one issue I reported meant the code wouldn't link on Windows 11

**Comment by neheb** [comment]
Found an issue on big endian with the new video code. I don't think it's a big enough issue to hold back any release.

**Comment by benmccann** [comment]
Good find. I agree it should not be a blocker as most users will be little endian and it's not a regression since it's in new functionality.

**Comment by pkrause1980** [comment]
Can issue #2206 be considered.? Thanks!

**Comment by nehaljwani** [comment]
**Exiv2 vT.B.D 2023-05-TBD**

TODO: Add milestone information (or get rid of this section)
| Topic          | More Info |
|:--             |:--        | 

**Exiv2 vT.B.D Acknowledgements**

According to `git shortlog -s -n v0.27.6..main`
| Commits          | Author |
|:--             |:--        | 
| 433 | Rosen Penev
| 417 | Kevin Backhouse
| 368 | Luis Díaz Más
|  92 | Miloš Komarčević
|  87 | Peter S
|  72 | Christoph Hasse
|  52 | Robin Mills
|  46 | Mohamed Ali Chebbi
|  37 | Luis Diaz
|  36 | Alex Esseling
|  24 | Peter
|  19 | Luis Diaz Mas
|  18 | Viktor Schneider
|  15 | norbertwg
|   8 | Sturmflut
|   7 | Alexander Steffen
|   7 | Michael Allman
|   5 | Jim Easterbrook
|   5 | dependabot[bot]
|   4 | Dmitry Fedorov
|   4 | Gribouilleuse
|   3 | Colin Watson
|   3 | mergify[bot]
|   2 | Alan01
|   2 | David Houlder
|   2 | Evan Miller
|   2 | Heiko Bauke
|   2 | Peter Kovář
|   2 | tbeu
|   1 | Andrea Giudiceandrea
|   1 | Antonio Rojas
|   1 | Attila Oláh
|   1 | Ben McCann
|   1 | D4N
|   1 | Daniel Vogelbacher
|   1 | Eli Schwartz
|   1 | Fabrice Fontaine
|   1 | Facundo
|   1 | Jakub Wilk
|   1 | Jeka Pats
|   1 | John55h
|   1 | Josh Soref
|   1 | Matthias Kuhn
|   1 | Moshe Kaplan
|   1 | Shawn Baker
|   1 | Steve Robbins
|   1 | andyrtr
|   1 | codesee-maps[bot]
|   1 | mousepotato
|   1 | nulllinie

**Exiv2 vT.B.D Release Notes (updated 2023-05-01)**

|  Group        | PR     |  Topic                                        | Reference  |
|:---           |:--     |:--                                            |:--     |
|**Documentation**|#1564|Update security process|
||#1892|Update SECURITY.md |
||#1872|Update exiv2 program and manpage|
||#1905|Update website docs|
||#1914|Update manpage and use in related files|
||#1964|Fix doc description of Image::iccProfileDefined()|#1953
||#2034|Updating documentation on main to respect ctest|
||#2043|Update test file location in manpage|
||#2070|Fix branch names in CONTRIBUTING.md (master was renamed to main)|
||#2075|Fix branch names in GIT_GUIDELINES.md (master was renamed to main)|
||#2076|Update markdown documentation|
|**Build/CI/Fuzz/CodeScan/CodeCov/GHA/Tests**|#1546|Fix CI for Ubuntu 18.04|
||#1559|Fix Cmake Build Type|#1558
||#1560|Re-enable Sanitizers Build|
||#1756|Add BUILD_WITH_STACK_PROTECTOR option|
||#1768|Add custom CodeQL query for null iterator deref|
||#1718|Fix building for ios|
||#1775|Fix code scanning alert: "Multiplication result converted to larger type"|
||#1771|CodeQL query to detect unsafe uses of std::vector::operator[] |
||#1773|Add fuzz target|
||#1781|Ignore results in the xmpsdk directory|
||#1804|Add fuzzing dictionary and test all printStructure() options|
||#1813|Add sanitizer flags when fuzzing|
||#1565|Fix Include Path|#1516
||#1869|Decouple EXIV2_BUILD_FUZZ_TESTS from EXIV2_TEAM_USE_SANITIZERS|
||#1875|Add special build mode for OSS-Fuzz|
||#1876|Add OSS-Fuzz status badge and CIFuzz integration|
||#1883|Only build XMLValidator when XMP is enabled|
||#1898|Standardize on conan 1.39.0|
||#1899|Add doc to release workflow|
||#1806|Call Metadatum::print() to increase fuzzing coverage|
||#1900|Remove unused constructor that was causing a build failure in OSS-Fuzz|
||#1908|Annotate some strings with xgettext:no-c-format|
||#1915|Pass -fstack-protector-strong to linker |
||#1907|Add custom update-pot target|
||#1940|Add workaround for conan outage|
||#1942|Fix Conan on macOS|
||#1943|Fix Conan on macOS (another one)|
||#1927|Add second test to improve code coverage|
||#1962|Fix macOS build failure|
||#1976|Try a newer version of Conan.|
||#1986|Remove Conan from MacOS build|
||#1988|Add/Update git templates from old-master|
||#1987|Remove opensuse/tumbleweed|
||#2035|Forward CI changes|
||#2049|Rename GHA workflows for having more intuitive navigation|
||#2060|Extend unit tests for BMP & Datasets|
||#2061|Adding static analysis with PVS-Studio|
||#2080|Revise python test and dependencies to sample applications|
||#2086|Fix Release pipeline|
||#2098|Remove PR-Status badges from README|
||#2106|Install new image_types.h header|#2105
||#2101|Linux distros jobs on PRs + Fix linking issues on some platforms|#2100
||#2187|Produce releases with Interprocedural optimisations (LTO) |
||#2240|Fix CI jobs - Update conan packages|
||#2242|Fix special builds for main|
||#2273|Eliminate some false positive results|
||#2314|Upgrade conan to version 1.51.0 |
||#2361|Bump conan version to 1.52.0 |
||#2419|improved in-progress job canceling|
||#2426|Remove deprecated option in conan|
||#2431|Enable Workflow dispatch on Windows CI jobs|
||#2433|Replace cygwin github action|
||#2437|Install the CodeSee workflow|
||#2457|Add inih dependency to release workflow|
||#2465|Add EXIV2_ENABLE_INIH flag to enable building without libinih|
||#2468|inih now available for Cygwin|
||#2471|Improve pkg-config file generation|
||#2536|meson updates|
||#2540|update conan|
||#2587|Add test case (complements #2567)|
|**Bugs/Security Fixes**|#1612|Add bounds check in Jp2Image::doWriteMetadata()|https://github.com/Exiv2/exiv2/security/advisories/GHSA-7569-phvm-vwc2
||#1621|Fix infinite loop caused by subBox with zero size|
||#1714|Fix compilation with -Wunused-variable|
||#1767|Fix null iterator deref in printCsLensTypeByMetadata|#1763
||#1774|Avoid processing MOV (quicktime) files when BMFF is enabled|#1724
||#1780|Fix infinite loop in Image::printIFDStructure|https://github.com/Exiv2/exiv2/security/advisories/GHSA-m479-7frc-gqqg
||#1786|Safer std::vector indexing |https://github.com/Exiv2/exiv2/security/advisories/GHSA-v5g7-46xf-h728
||#1794|Enforce BMFF box nesting|#1793
||#1787|Extra checking to prevent loop counter from wrapping around|https://github.com/Exiv2/exiv2/security/advisories/GHSA-hqjh-hpv8-8r9p
||#1797|Make sure that read is complete to prevent infinite loop|https://github.com/Exiv2/exiv2/security/advisories/GHSA-9jh3-fcc3-g6hv
||#1798|&bytes[0] (std::vector) will crash if bytes has zero elements|https://github.com/Exiv2/exiv2/security/advisories/GHSA-g44w-q3vm-gwjq
||#1799|bufRead needs to be adjusted after seek() |https://github.com/Exiv2/exiv2/security/advisories/GHSA-mvc4-g5pv-4qqq
||#1822|Add bounds-check to prevent out-of-bounds read in memcmp|#1815
||#1825|jp2image.cpp: check size before allocation to avoid out-of-memory errors|#1812
||#1823|Fix memory leak in pngimage.cpp |#1817
||#1826|Check that the string isn't empty|#1819
||#1832|Check value is in range before casting from double to uint32_t|#1827
||#1836|Replace assertion with an error message|#1833
||#1837|Safer casting from double to long in value.hpp |#1830
||#1844|Use DataBuf rather than raw malloc in WebPImage::decodeChunks|#1841
||#1843|Check that the float is within the range of an int before casting|#1838
||#1852|Replace assertion with error message in TiffDirectory::writeDirEntry() |#1845
||#1849|Replace assertion with an error message|#1847
||#1857|Fix compiler warning on Apple/M1/Clang|#1856
||#1853|Add test for issue 1821 |#1821
||#1878|Add some XML validation to avoid xmpsdk bugs|#1877
||#1824|Corrected JPEG XL file type|
||#1882|Throw an error if the size of the preview is greater than 1MB|#1881
||#1888|Avoid reading 1 byte off the end when the string does not contain a '\0' byte|#1887
||#1886|Make fields of DataBuf private|
||#1910|Fix quadratic loops in pentaxmn_int.cpp|#1909
||#1902|Comment out bogus code in XMPUtils.cpp|#1901
||#1913|Fix integer overflow in print0x0007|#1912
||#1918|fix out of range access, minor performance improvement|https://bugs.launchpad.net/ubuntu/+source/exiv2/+bug/1942799
||#1932|Update canonmn_int.cpp to correct lens "Canon EF 80-200mm f/4.5-5.6 II"|#1906
||#1921|Fix UBSAN failure caused by left-shift of negative number|#1920
||#1947|Hide "static constexpr auto" from SWIG|
||#1955|Fix integer overflow in leap year calculation|#1954
||#1979|Add exiv2 extract thumb to stdout|#1934
||#2000|Fix exiv2: verbose extract to stdout|#1934
||#1999|Restore modified tags after TZ manipulation|#1998
||#2015|Treat Exif.Sony1.PreviewImage as undefined tag|#2001
||#2007|Fix integer overflow in PanasonicMakerNote::printAccelerometer|#2006
||#2028|Use memmove in TiffEncoder::updateDirEntry|#2027
||#2039|Fix bug in loop. It wasn't iterating over the elements of dateStrings as intended|#2036
||#1929|Fix exiv2 -pR for Sony images |#1805
||#2053|Update XMP tags for IPTC 2021.1|#1959
||#2078|"Revert ""Delete dead code"""|
||#2090|Get argv from main entry point encoded as UTF-8 |#1996
||#2111|Catch std::exception in iptcprint|
||#1919|fix for reading jpeg-xr images|
||#2133|Strip XMP raw packet before decoding|#2126
||#2137|Recover IptDataSets::recordId in the API|
||#2146|DateValue is now a bit more permissive with malformed dates|#2144
||#2161|Use std::round() in float_to_integer_helper|#2160
||#2155|Fix in Jp2 metadata writing & improvements in reading (2) |#2147
||#2180|Fix out of bounds read in isValidBoxFileType()|#2178
||#2191|Detect integer-overflow and throw in that case|#2190
||#2181|Fix integer overflow|#2179
||#2234|Account for header bytes for Exif and XMP boxes|#2233
||#2244|More bounds checking in Adjust::adjustDateTime|
||#2254|PNG: always strip the existing iCCP chunk|
||#2248|fix naming of canon EF 35-80mm.|#2247
||#2272|Replace assert with enforce|#2268
||#2275|"Fix some ""unsafe vector access"" warnings"|
||#2321|Fix integer overflow in printDegrees|#2320
||#2347|Avoid potential integer overflow in QuickTimeVideo::userDataDecoder|
||#2348|Don't allow zero width/height in WebPImage::inject_VP8X|#2270
||#2346|Fix buffer overflow in QuickTimeVideo::tagDecoder|#2345
||#2378|Fix bugs in QuickTimeVideo::userDataDecoder|#2376, #2377
||#2384|Avoid null pointer deref|#2383
||#2394|stop looping when eof is hit|#2393
||#2424|Fix long-running loop in QuickTimeVideo::sampleDesc|#2423
||#2428|Fix memory leak in BmffImage::brotliUncompress|#2427
||#2440|Prefer writing PNG eXIf chunk|#2250
||#2466|Silence unused parameter warning when EXV_ENABLE_INIH is undefined|
||#2479|Export QuickTimeVideo|
||#2509|check that block is not corrupted before decoding|
||#2533|Fix long-running loop|
||#2534|Fix integer overflow which enables enforce to be bypassed|
||#2544|Check that the pointer isn't NULL|
||#2551|Minor refactorings to stop UBSAN from complaining|#2539
||#2585|Uppercase Pentax to PENTAX at lines 819 to 822|
|**Library Features/Improvements**|#1554|Enable BMFF by default|
||#1551|Improve printing of GPS info|#1541
||#1738|Add –lint to exifprint|
||#1772|Add SonyMisc1 (Tag 0x9403) makernote tags|#280
||#1692|New Canon Lens Identification + Automatic Test of all Lenses|
||#1777|Update Sony2Fp tags|
||#1548|Add Exif Gamma tag|
||#1792|Add SonyMisc2b makernote tags (tag 9404) |
||#1800|Add SonyMisc3c makernote tags (tag 9400) |
||#1922|Add SonyMisc(2b\|3c) Groups for Sony TIFF files|
||#1926|Add conditions to 2 Sony temperature tags|
||#1944|Extending Nikon3.ColorBalance version coverage|
||#1974|BMFF Performance boost: don't read boxes we're not interested in|#1961
||#1978|Limit CR3 previews to JPEG only|#1893
||#1963|Add NikonFl7 makernotes group|#1941
||#1967|Add readOrThrow and seekOrThrow to BasicIo|
||#2069|Add Sigma 12-24mm F4 DG HSM Art lens for Nikon|
||#2073|Add Tokina AT-X 14-20 F2 PRO DX lens|
||#2093|New ImageTypes enum class|
||#2087|Update canon tags 2 |
||#2110|"Make Nikon ContrastDetectAFInFocus tags Bytes that print as ""Yes"" or ""No"""|
||#2071|Add instructions for adding new Nikon F mount lenses|
||#2172|Initial support for OM System MakerNote|#2126
||#2203|Create da.po|
||#2215|Add some F mount lenses|#2193, #2196, #2214
||#1949|add support for more fujifilm tags|
||#2374|several keys added to easyaccess|#2338
||#2381|Support brotli compressed boxes in JPEG XL|
||#2216|Add more Canon RF lenses|
||#2417|Add Canon R6m2 ID|
||#2444|Lens Recognition HD PENTAX-DA* 11-18mm F2.8 ED DC AW|#2388
||#2443|Remove libinih from codebase and add it as a dependency instead|#1811
||#2461|Delete contrib/organize directory|
||#2458|Rework Asf, Riff video, test data and fix OSS-Fuzz issues|
||#2422|Interpretation for Exif.Photo.LensSpecification|
||#2582|Update Sony file format table|#2569
||#2472|Fix shared object versioning|
|**Code Improvements**|#1555|C++ Modernization: auto_ptr→unique_ptr|
||#1874|remove most usages of std::make_pair|
||#1930|Clarify ownership model of CiffComponent::pData_|
||#1956|Use std::move to transfer ownership of DataBufs|
||#2040|Use std::shared_ptr for storage in TiffEntryBase|
||#2063|Modernise code around iterators & smart pointers|
||#2099|Separate exiv2 app source files from exiv2lib src folder|
||#2107|Replace Metadatum::toLong() with Metadatum::toInt64()|
||#2109|Refactoring & cleanup|
||#2118|Refactoring & Cleanup (Moving to size_t usage & less naked new/deletes)|
||#2124|mostly clang-tidy cleanups|
||#2125|some clang-tidy|
||#2127|use make_unsigned_t|
||#2130|misc stuff|
||#2129|conversions to make_unique|
||#2132|misc cleanups|
||#2135|manual nullptr removals|
||#2131|Include what you use + more SPDX identifiers + few other cleanups|
||#2141|Improvements in the Error class|#2116
||#2145|More migration to size_t|
||#2152|clang-format whole project + new CI workflow for checking style|
||#2163|Another cleanup round in main|
||#2158|array and constexpr conversions|
||#2169|remove various usages of memset|
||#2175|mostly nodiscard|
||#2177|clang-tidy and manual stuff|
||#2198|some clang-tidy|
||#2200|clang-tidy with headers|
||#2197|Refactoring in JpegImage and Photoshop classes|
||#2211|clang-tidy: default member init|
||#2212||
||#2209|Improvements around TODO comments|
||#2245|clang-tidy 14 fixes|
||#2231|types cleanup exiv2app.hpp|
||#2251|Change return type of BasicIo::tell() to size_t|
||#2266|"Fix some ""signed shift"" warnings"|
||#2252|clang-tidy: use default member init|
||#2271|"Fix some ""signed shift"" warnings"|
||#2267|mostly clang-tidy fixes|
||#2277|Simplify the iptcDataSegs logic in jpgimage.cpp|
||#2276|Move enums from tags_int.hpp to tags.hpp|
||#2296|Use size_t for the offset argument in TiffComponent|
||#2283|Convert type of depth parameter to size_t|
||#2390|Use Safe::add() in tiffvisitor_int.cpp|
||#2504|fixes|
||#2548|add various moves |

I've tried to categorize commits based on relevance to the closest group.

There might be some commits that have not been included here. I'll try to add them later if I find any.

If I've failed to acknowledge anyone's contribution, I apologize. Please let me know and I'll update this comment.

I'll attempt to cut a release based on the state of votes at https://github.com/Exiv2/exiv2/issues/2406#issuecomment-1511936787, on the weekend following Cinco De Mayo. Please go through the changelog (maybe it will make you reconsider your vote 😉) .

**Comment by nehaljwani** [comment]
@piponazo @hassec Could you please submit your vote in https://github.com/Exiv2/exiv2/issues/2406#issuecomment-1511936787 as well?

**Comment by nehaljwani** [comment]
Exiv2 v0.28.0 has been released.

**Comment by kevinbackhouse** [comment]
Thanks @nehaljwani!
